src/gt4py/storage/storage.py

```diff
@@ -14,604 +14,1112 @@
+            else:
+                storage_t = GPUStorage
+                kwargs = {}
+        else:
+            storage_t = CPUStorage
+            kwargs = {}
+
+        assert gt_utils.is_iterable_of(shape, item_class=int, iterable_class=tuple)
+        assert gt_utils.is_iterable_of(mask, item_class=bool, iterable_class=tuple)
+
+        return storage_t.__new__(
+            storage_t,
+            array=data,
+            shape=shape,
+            dtype=dtype,
+            default_origin=default_origin,
+            mask=mask,
+            layout_map=layout_map,
+            alignment=alignment,
+            copy=copy,
+            **kwargs,
         )
-        obj._backend = backend
-        obj.is_stencil_view = True
-        obj._mask = mask
-        obj._check_data()
 
-        return obj
-
-    @property
-    def backend(self):
-        return self._backend
+    # def __init__(
+    #     self,
+    #     *,
+    #     shape,
+    #     dtype,
+    #     data=None,
+    #     default_origin=None,
+    #     mask=None,
+    #     default_parameters=None,
+    #     alignment=None,
+    #     device=None,
+    #     layout_map=None,
+    #     copy=True,
+    # ):
+    #     # """
+    #     #
+    #     # Parameters
+    #     # ----------
+    #     #
+    #     # shape: tuple of ints
+    #     #     the shape of the storage
+    #     #
+    #     # dtype: data type compatible with numpy dtypes
+    #     #     supported are the floating point and integer dtypes of numpy
+    #     #
+    #     # default_parameters: string, backend identifier
+    #     #     Currently possible: debug, numpy, gtx86, gtmc, gtcuda
+    #     #     specifies according to which backend alignment, layout_map and device should be chosen if not otherwise
+    #     #     specified.
+    #     # default_origin: tuple of ints
+    #     #     determines the point to which the storage memory address is aligned.
+    #     #     for performance, this should be the coordinates of the most common origin
+    #     #     at call time.
+    #     #     when calling a stencil and no origin is specified, the default_origin is used.
+    #     #
+    #     # mask: list of booleans
+    #     #     False entries indicate that the corresponding dimension is masked, i.e. the storage
+    #     #     has reduced dimension and reading and writing from offsets along this axis acces the same element.
+    #     # """
+    #
+    #     self._alignment = alignment
+    #     self._layout_map = layout_map
+    #     self._mask = mask
+    #     self._default_origin = default_origin
+
+    def all(*args, **kwargs):
+        return np.all(*args, **kwargs)
+
+    def any(*args, **kwargs):
+        return np.any(*args, **kwargs)
+
+    def min(*args, **kwargs):
+        return np.min(*args, **kwargs)
+
+    def max(*args, **kwargs):
+        return np.max(*args, **kwargs)
+
+    def transpose(*args, **kwargs):
+        return np.transpose(*args, **kwargs)
```

Comment:
It may be worth adding some basic tests that these methods return what you expect, e.g. that if numpy is updated in a way that this turns your storages to ndarray, you find out about it from a failing test.